runtime/vm/JFRChunkWriter.cpp

```diff
@@ -815,4 +825,48 @@ VM_JFRChunkWriter::writeInitialSystemPropertyEvents(J9JavaVM *vm)
 	}
 }
 
+void
+VM_JFRChunkWriter::writeInitialEnvironmentVariableEvents()
+{
+	J9SysinfoEnvIteratorState envState;
+	memset(&envState, 0, sizeof(envState));
+
+	/* Call init with zero length buffer to get the required buffer size */
+	int32_t result = j9sysinfo_env_iterator_init(&envState, NULL, 0);
+
+	if (result >= 0) {
+		int32_t bufferSize = result;
+		void *buffer = j9mem_allocate_memory(bufferSize, OMRMEM_CATEGORY_VM);
+
+		if (NULL != buffer) {
+			J9SysinfoEnvElement envElement = {NULL};
+
+			result = j9sysinfo_env_iterator_init(&envState, buffer, bufferSize);
```

Comment:
need to check if `if (result < 0) {` if so there was a problem getting the env vars

Reply:
Oh okay, miseed that. Leave it as is

Reply:
Actaully, sorry, an error can still occur even if the buffer is valid, so please add the check